ui/js/jquery.pods.js

```diff
@@ -1849,6 +1849,65 @@
             }
         };
 
+    //--!! Ugly proof of concept code
+
+    /**
+     * Close the modal on form submit, if one is open
+     */
+    $('form#post').on( 'submit', function( event ) {
+        var $_GET = {};
+        if(document.location.toString().indexOf('?') !== -1) {
+            var query = document.location
+                .toString()
+                // get the query string
+                .replace(/^.*?\?/, '')
+                // and remove any existing hash string (thanks, @vrijdenker)
+                .replace(/#.*$/, '')
+                .split('&');
+
+            for(var i=0, l=query.length; i<l; i++) {
+                var aux = decodeURIComponent(query[i]).split('=');
+                $_GET[aux[0]] = aux[1];
+            }
+        }
+        if ( $_GET[ 'pods_modal' ] || 0 ) {
+            $( '.tb-close-icon', parent.document ).click();
+        }
+    } );
+
+
+    /**
+     * Modal display and ajax updates
+     */
+    /*global pods_relationship_popup_data */
+    $( '.pods-related-edit' ).on( 'click', function( e ) {
+
+        var field_name = $( this ).data( 'field-name' );
+        var popup_data = pods_relationship_popup_data[ field_name ];
+
+        e.preventDefault();
+        tb_show( '', popup_data.url );
+
+        // Fired when the popup unloads
+        $( '#TB_window' ).on( "tb_unload", function () {
+
+            var data = {
+                'action':          'pods_relationship_popup',
+                'field_type':      popup_data.field_type,
+                'options':         popup_data.options,
+                'id':              popup_data.id,
+                'name':            popup_data.name,
+                'value':           popup_data.value
```

Comment:
It does introduce a new area of ambiguity, however.  At the time of an ajax refresh there could be a difference between the state of the record as saved in the DB and change you may have made that are only in the DOM (haven't saved yet).  New things to consider.